### ISSUE TITLE
Extended config

### ISSUE THREAD

**Issue by mozey** [mozey/config#47]
<s>Similar to `APP_TEMPLATE`, consider another special prefix: `APP_X` (extension)

If there is a sub-dir path matching the value, then the config file in that dir will be included.</s>

Keys in the extended config must not duplicate keys in the main config

Related issues
- https://github.com/mozey/config/issues/37

**Comment by mozey** [comment]
This feature is useful for extensions or plugins defined in packages that are sub-directories of APP_DIR

The main config file is an index of sub-directories that also contain config files

The extended config is only loaded if the path exists. 

For example, if the package config is `APP_X_FOO=path/to/foo`, then
```go
conf, err = config.LoadFile("dev")
if err != nil {
  // Err if path does not exist
}

// Returns true if path exists.
// Useful to check if an extension is enabled
conf.XFoo()
```

To disable loading extended config, prefix the value with a dash, e.g. `APP_X_FOO=-path/to/foo`

**Comment by mozey** [comment]
Closes #37

**Comment by mozey** [comment]
Flag for loading extended config
- Extent config from first parent dir with config file: -x 
- Extend config from specific sub dir, support multiple: -x path/to/ext 

Generate code for extended config inside the extension package, maintain a clear separation between config files.

Main config does not need to know about extensions, and extensions do not specify which main config to use.

Compose config by nesting APP_DIRs.

Duplicate keys result in an error, use a unique prefix for each extension.

Use cases
- CLI usage and testing inside extension may use config from parent
- [Base64 config](https://github.com/mozey/config/issues/28) may include extensions
